### ISSUE TITLE
Unable to achieve 76.6% accuracy

### ISSUE THREAD

**Issue by nabeel3133** [xiaohong1/COVID-ViT#5]
I managed to run your code and start the training on the pre-trained model however, I am getting the same results (about 50% accuracy) as shown in the jupyter notebook
```
 0%|          | 0/755 [00:00<?, ?it/s]
Epoch : 1 - loss : 0.6947 - acc: 0.4957 - val_loss : 0.6930 - val_acc: 0.5084

  0%|          | 0/755 [00:00<?, ?it/s]
Epoch : 2 - loss : 0.6947 - acc: 0.4881 - val_loss : 0.6930 - val_acc: 0.5084

  0%|          | 0/755 [00:00<?, ?it/s]
Epoch : 3 - loss : 0.6948 - acc: 0.5003 - val_loss : 0.6942 - val_acc: 0.5088

  0%|          | 0/755 [00:00<?, ?it/s]
Epoch : 4 - loss : 0.6941 - acc: 0.5060 - val_loss : 0.6931 - val_acc: 0.5088

  0%|          | 0/755 [00:00<?, ?it/s]
Epoch : 5 - loss : 0.6951 - acc: 0.4868 - val_loss : 0.6934 - val_acc: 0.5093

  0%|          | 0/755 [00:00<?, ?it/s]
Epoch : 6 - loss : 0.6944 - acc: 0.5146 - val_loss : 0.6936 - val_acc: 0.4912

  0%|          | 0/755 [00:00<?, ?it/s]
Epoch : 7 - loss : 0.6947 - acc: 0.4924 - val_loss : 0.6935 - val_acc: 0.4907

  0%|          | 0/755 [00:00<?, ?it/s]
Epoch : 8 - loss : 0.6949 - acc: 0.4954 - val_loss : 0.6930 - val_acc: 0.5093

  0%|          | 0/755 [00:00<?, ?it/s]
Epoch : 9 - loss : 0.6945 - acc: 0.5010 - val_loss : 0.6966 - val_acc: 0.4921

  0%|          | 0/755 [00:00<?, ?it/s]
Epoch : 10 - loss : 0.6949 - acc: 0.4874 - val_loss : 0.6934 - val_acc: 0.5093

  0%|          | 0/755 [00:00<?, ?it/s]
Epoch : 11 - loss : 0.6941 - acc: 0.5056 - val_loss : 0.6971 - val_acc: 0.5084

  0%|          | 0/755 [00:00<?, ?it/s]
Epoch : 12 - loss : 0.6946 - acc: 0.5023 - val_loss : 0.6949 - val_acc: 0.4907

  0%|          | 0/755 [00:00<?, ?it/s]
Epoch : 13 - loss : 0.6945 - acc: 0.4954 - val_loss : 0.6933 - val_acc: 0.4916

  0%|          | 0/755 [00:00<?, ?it/s]
Epoch : 14 - loss : 0.6942 - acc: 0.5030 - val_loss : 0.6958 - val_acc: 0.4907

  0%|          | 0/755 [00:00<?, ?it/s]
Epoch : 15 - loss : 0.6935 - acc: 0.5126 - val_loss : 0.6965 - val_acc: 0.5079

  0%|          | 0/755 [00:00<?, ?it/s]
Epoch : 16 - loss : 0.6957 - acc: 0.4967 - val_loss : 0.6935 - val_acc: 0.4907

  0%|          | 0/755 [00:00<?, ?it/s]
Epoch : 17 - loss : 0.6941 - acc: 0.5023 - val_loss : 0.6932 - val_acc: 0.5088

  0%|          | 0/755 [00:00<?, ?it/s]
Epoch : 18 - loss : 0.6948 - acc: 0.4973 - val_loss : 0.6930 - val_acc: 0.5084

  0%|          | 0/755 [00:00<?, ?it/s]
Epoch : 19 - loss : 0.6936 - acc: 0.5053 - val_loss : 0.6957 - val_acc: 0.4912

  0%|          | 0/755 [00:00<?, ?it/s]
Epoch : 20 - loss : 0.6945 - acc: 0.4904 - val_loss : 0.6934 - val_acc: 0.5079

  0%|          | 0/755 [00:00<?, ?it/s]
Epoch : 21 - loss : 0.6943 - acc: 0.4940 - val_loss : 0.6931 - val_acc: 0.5088

  0%|          | 0/755 [00:00<?, ?it/s]
Epoch : 22 - loss : 0.6950 - acc: 0.4957 - val_loss : 0.6941 - val_acc: 0.5093

  0%|          | 0/755 [00:00<?, ?it/s]
Epoch : 23 - loss : 0.6942 - acc: 0.4930 - val_loss : 0.6937 - val_acc: 0.4912

  0%|          | 0/755 [00:00<?, ?it/s]
Epoch : 24 - loss : 0.6942 - acc: 0.4950 - val_loss : 0.6930 - val_acc: 0.5079

  0%|          | 0/755 [00:00<?, ?it/s]
Epoch : 25 - loss : 0.6947 - acc: 0.4957 - val_loss : 0.6930 - val_acc: 0.5079

  0%|          | 0/755 [00:00<?, ?it/s]
Epoch : 26 - loss : 0.6939 - acc: 0.4904 - val_loss : 0.6972 - val_acc: 0.5084

  0%|          | 0/755 [00:00<?, ?it/s]
Epoch : 27 - loss : 0.6941 - acc: 0.5070 - val_loss : 0.6930 - val_acc: 0.5088

  0%|          | 0/755 [00:00<?, ?it/s]
```

Can you let me know what changes are required to be done to achieve 76.6% accuracy as mentioned in the paper?

**Comment by xiaohong1** [comment]
If you are running 3D model, the accuracy might be lower since the lesioned regions/sub-volumes tend to be in a very small proportion in comparison with the whole  3D volume. This is why 2D models are better. To improve it, combine both 2D and 3D together. Have not tried on this data set yet, but it worked for 2D/3D brain images.

**Comment by nabeel3133** [comment]
So, the reported 76.6% accuracy for ViT in the paper is for 2D model?
